File: python/katas/fizzbuzz.py
from typing import List


def fizzbuzz(start: int, end: int) -> List[str]:
    return [convert(x) for x in range(start, end + 1)]


def convert(n: int) -> str:
    if n % 3 == 0 and n % 5 == 0:
        return "fizzbuzz"
    elif n % 5 == 0:
        return "buzz"
    elif n % 3 == 0:
        return "fizz"
    else:
        return f"{n}"

File: elixir/lib/fizzbuzz.ex
defmodule Fizzbuzz do
  defp divisible(d, n), do: rem(n, d) == 0

  defp convert(n) do
    cond do
      divisible(5, n) and divisible(3, n) -> "fizzbuzz"
      divisible(5, n) -> "buzz"
      divisible(3, n) -> "fizz"
      true -> n
    end
  end

  @spec fizzbuzz(integer, integer) :: list
  def fizzbuzz(start, finish) do
    for n <- start..finish, do: convert(n)
  end
end
